java/integTest/src/test/java/glide/SharedCommandTests.java

```diff
@@ -12922,4 +12922,132 @@ public void wait_timeout_check(BaseClient client) {
                     () -> testClient.wait(100L, 0L).get(1000, TimeUnit.MILLISECONDS));
         }
     }
+
+    @SneakyThrows
+    @ParameterizedTest(autoCloseArguments = false)
+    @MethodSource("getClients")
+    public void xinfoStream(BaseClient client) {
+        String key = UUID.randomUUID().toString();
+        String groupName = "group" + UUID.randomUUID();
+        String consumer = "consumer" + UUID.randomUUID();
+        String streamId0_0 = "0-0";
+        String streamId1_0 = "1-0";
+        String streamId1_1 = "1-1";
+
+        // Setup: add stream entry, create consumer group and consumer, read from stream with consumer
+        final LinkedHashMap<String, String> dataToAdd =
+                new LinkedHashMap<>() {
+                    {
+                        put("a", "b");
+                        put("c", "d");
+                    }
+                };
+        assertEquals(
+                streamId1_0,
+                client.xadd(key, dataToAdd, StreamAddOptions.builder().id(streamId1_0).build()).get());
+        assertEquals(OK, client.xgroupCreate(key, groupName, streamId0_0).get());
+        client.xreadgroup(Map.of(key, ">"), groupName, consumer).get();
+
+        Map<String, Object> result = client.xinfoStream(key).get();
+        assertEquals(1L, result.get("length"));
+        Object[] expectedFirstEntry = new Object[] {streamId1_0, new String[] {"a", "b", "c", "d"}};
+        assertDeepEquals(expectedFirstEntry, result.get("first-entry"));
+
+        // Only one entry exists, so first and last entry should be the same
+        assertDeepEquals(expectedFirstEntry, result.get("last-entry"));
+
+        // Call XINFO STREAM with a byte string arg
+        Map<GlideString, Object> result2 = client.xinfoStream(gs(key)).get();
+        assertEquals(1L, result2.get(gs("length")));
+        Object[] gsFirstEntry = (Object[]) result2.get(gs("first-entry"));
+        Object[] expectedFirstEntryGs =
+                new Object[] {gs(streamId1_0), new Object[] {gs("a"), gs("b"), gs("c"), gs("d")}};
+        assertArrayEquals(expectedFirstEntryGs, gsFirstEntry);
+
```

Comment:
Do you test non-GS version of new command?

Reply:
Sorry what do you mean?

Reply:
It's included in the tests above.